src/s1reader/s1_burst_slc.py

```diff
@@ -579,6 +585,332 @@ def az_carrier_components(self, offset, position):
+
+        # convert azimuth time to seconds from the reference epoch of burst_in.orbit
+        vec_aztime_fm_rate_sec = np.zeros(self.extended_coeffs.vec_aztime_fm_rate.shape)
+        for i, datetime_vec in enumerate(self.extended_coeffs.vec_aztime_fm_rate):
+            vec_aztime_fm_rate_sec[i] = (isce3.core.DateTime(datetime_vec)
+                                         - self.orbit.reference_epoch).total_seconds()
+
+        vec_aztime_dc_sec = np.zeros(self.extended_coeffs.vec_aztime_dc.shape)
+
+        for i, datetime_vec in enumerate(self.extended_coeffs.vec_aztime_dc):
+            vec_aztime_dc_sec[i] = (isce3.core.DateTime(datetime_vec)
+                                    - self.orbit.reference_epoch).total_seconds()
+
+        # calculate splined interpolation of the coeffs. and tau_0s
+        interpolator_tau0_ka = InterpolatedUnivariateSpline(
+                                        vec_aztime_fm_rate_sec,
+                                        self.extended_coeffs.vec_tau0_fm_rate,
+                                        k=1)
+        tau0_ka_interp = interpolator_tau0_ka(vec_t)[..., np.newaxis]
+
+        interpolator_tau0_fdc_interp = InterpolatedUnivariateSpline(
+                                        vec_aztime_dc_sec,
+                                        self.extended_coeffs.vec_tau0_dc,
+                                        k=1)
+        tau0_fdc_interp = interpolator_tau0_fdc_interp(vec_t)[..., np.newaxis]
+
+        # add range time origin to vec_tau
+        grid_tau += tau0_ka_interp * np.ones(vec_tau.shape)[np.newaxis, ...]
+
+
+        # Interpolate the DC and fm rate coeffs along azimuth time
+        list_a_interp = [] # Doppler centroid
+        list_b_interp = [] # Fm rate
+        for i in range(3):
+            interpolator_coeff_a = InterpolatedUnivariateSpline(
+                                        vec_aztime_dc_sec,
+                                        self.extended_coeffs.lut_coeff_dc[:, i],
+                                        k=1)
+            list_a_interp.append(interpolator_coeff_a(vec_t))
+
+            interpolator_coeff_b = InterpolatedUnivariateSpline(
+                                        vec_aztime_fm_rate_sec,
+                                        self.extended_coeffs.lut_coeff_fm_rate[:, i],
+                                        k=1)
+            list_b_interp.append(interpolator_coeff_b(vec_t))
+
+        arr_coeff_a = np.array(list_a_interp).transpose()
+        arr_coeff_b = np.array(list_b_interp).transpose()
+
+        # Run topo on scratch directory
+        dem_raster = isce3.io.Raster(path_dem)
+        epsg = dem_raster.get_epsg()
+        proj = isce3.core.make_projection(epsg)
+        ellipsoid = proj.ellipsoid
+        grid_doppler = isce3.core.LUT2d()
+
+        Rdr2Geo = isce3.geometry.Rdr2Geo
+
+        # TODO Add parameters (# interation, threshold, etc.) to rdr2geo
```

Comment:
I think this comment can be removed. I guess that the number of iterations and threshold will come from the COMPASS runconfig for geocoded CSLC?

Reply:
Yes. The task described in the TODO above should be addressed in `COMPASS` side, not `s1-reader`.

I'll submit a PR to COMPASS to integrate the azimuth FM mismatch mitigation into COMPASS.